js/rect.js

```diff
@@ -22,13 +22,13 @@
         return this.bottom - this.top;
       },
       set width(v) {
-        this.right = this.left + v;
+        this.right = this.left + Number(v);
```

Comment:
I wonder if we should have a generic type enforcement for Rect that throws if any of the values is NaN. Also, there is already a Rect class on the Window object it seems (that is different from this one).